dotCMS/src/main/java/com/dotcms/business/CacheableEagerFactory.java

```diff
@@ -122,6 +136,21 @@ default Optional<V> find(final K key) throws DotDataException {
      */
     default Map<K, V> findAll() throws DotDataException {
 
+        // Check if shutdown is in progress - avoid database operations during shutdown
+        boolean shutdownInProgress = false;
```

Comment:
this looks such as aspect for an annotation

Reply:
> this looks such as aspect for an annotation

Possibly but adds more complexity and risk for the time being and would be a good follow up PR.